portal-kernel/src/com/liferay/portal/kernel/cache/transactional/TransactionalPortalCacheHelper.java

```diff
@@ -276,16 +312,56 @@ private static PortalCacheMap _popPortalCacheMap() {
 				ArrayList::new, false);
 	private static volatile Boolean _transactionalCacheEnabled;
 
+	private static class MVCCUncommittedBuffer extends UncommittedBuffer {
+
+		@Override
+		public void commit(boolean readOnly) {
+			_portalCacheCounters.compute(
+				_portalCacheName,
+				(key, portalCacheCounter) -> {
+					if (portalCacheCounter != _portalCacheCounter) {
+						commitByRemove = true;
+					}
+
+					if (readOnly && commitByRemove) {
+						return portalCacheCounter;
+					}
+
+					super.commit(readOnly);
+
+					return portalCacheCounter + 1;
```

Comment:
@tinatian this plus one is too aggressive. In case there is no writer, all tx are readonly, they should all be able to commit. But with this +1, only the first commiter can do it, everyone else will back off.

Fundamentally, this count should represent real modification counts, it should only bump when we really received a modification.

Reply:
@shuyangzhou Yes, you are right, I will think about this